src/topology/metric_space/basic.lean

```diff
@@ -439,6 +439,12 @@ theorem mem_sphere' : y ∈ sphere x ε ↔ dist x y = ε := by rw [dist_comm, m
 theorem ne_of_mem_sphere (h : y ∈ sphere x ε) (hε : ε ≠ 0) : y ≠ x :=
 by { contrapose! hε, symmetry, simpa [hε] using h  }
 
+theorem pos_of_mem_sphere (hy : y ∈ sphere x ε) : 0 ≤ ε :=
+dist_nonneg.trans_eq hy
```

Comment:
```suggestion
theorem pos_of_sphere_nonempoty : (sphere x ε).nonempty) → 0 ≤ ε :=
λ ⟨y, hy⟩, dist_nonneg.trans_eq hy
```
since the goal doesn't refer to `y`.

Reply:
The existing lemma matches `pos_of_mem_ball`, so I don't think I want to make this change in this PR. You could change all of them at once in a follow-up.